Artifacts/linux-vsts-build-agent/Artifactfile.json

```diff
@@ -0,0 +1,45 @@
+{
+  "$schema": "https://raw.githubusercontent.com/Azure/azure-devtestlab/master/schemas/2016-11-28/dtlArtifacts.json",
+  "title": "Azure Pipelines Agent for Linux",
+  "publisher": "Microsoft",
+  "description": "Downloads and installs the Azure Pipelines agent, registers with the specified Azure DevOps Services account, and adds the VM to the specified agent pool.",
+  "tags": [
+    "VSTS",
+    "Build",
+    "CI",
+    "Linux"
+  ],
+  "iconUri": "https://cdn.vsassets.io/content/icons/favicon.ico",
+  "targetOsType": "Linux",
+  "parameters": {
+    "adoAccount": {
+      "type": "string",
+      "displayName": "Azure DevOps Organization Name",
+      "description": "The name of the Azure DevOps organization to add the agent to. This is the value in your Azure DevOps URL: e.g. 'myorg' in https://dev.azure.com/myorg."
+    },
+    "adoPat": {
+      "type": "securestring",
+      "displayName": "Azure DevOps Personal Access Token",
+      "description": "A personal access token with permissions to add agents. It will only be used to register the agent. See https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/v2-linux?view=azure-devops for more information"
+    },
+    "adoPool": {
+      "type": "string",
+      "displayName": "Agent Pool",
+      "description": "The agent pool this build agent should be added to."
+    },
+    "agentPath": {
+      "type": "string",
+      "displayName": "Agent Path",
+      "description": "Path where agent is stored. Any storage space used by the agent is in a subdirectory of the agent path."
+    },
+    "agentName": {
+        "type": "string",
+        "displayName": "Agent Name",
+        "description": "A unique name identifying this agent. If not specified, it defaults to the hostname of the machine running the agent",
+        "allowEmpty": true
+    }
+  },
+  "runCommand": {
+    "commandToExecute": "[concat('python3 ', './agent-install.py', ' --agent_path ''', parameters('agentPath'), ''' --ado_pat ''', parameters('adoPat'), ''' --ado_account ''', parameters('adoAccount'), ''' --ado_pool ''', parameters('adoPool'), ''' --agent_name ''', parameters('agentName'), '''')]"
```

Comment:
The `runCommand` seems to have an extra `'` at the end, like `''''`. Can you confirm this doesn't break the artifact?

Reply:
Here's what success looks like:
![success](https://user-images.githubusercontent.com/45858683/63376971-db3ddb00-c343-11e9-8b84-5f3216ce841a.png)

And here's the failure message when using an invalid personal access token:

{"status":"Failed","error":{"code":"ResourceDeploymentFailure","message":"The resource operation completed with terminal provisioning state 'Failed'.","details":[{"code":"VMExtensionProvisioningError","message":"VM has reported a failure when processing extension 'customScriptArtifact-763564051'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=1\n[stdout]\nfetching https://edgetools.visualstudio.com/_apis/distributedtask/packages/agent/linux-x64?$top=1&api-version=3.0 to determine agent package url\n\n[stderr]\nTraceback (most recent call last):\n File \"./agent-install.py\", line 107, in

Reply:
The four quotes in a row get resolved as: 1: Start of string, 2-3: escaped single quote, 4: End of string